src/openforms/submissions/models.py

```diff
@@ -150,6 +158,18 @@ def refresh_from_db(self, *args, **kwargs):
     def is_completed(self):
         return bool(self.completed_on)
 
+    def remove_sensitive_data(self):
```

Comment:
Please add an explicit `@transaction.atomic()` on this, we really don't want this to fail somewhere half-way.